internal/pkg/api/server/buffer/dependencies/store_test.go

```diff
@@ -52,14 +47,88 @@ func TestConfigStore_CreateReceiver(t *testing.T) {
 
 	found := false
 	for _, v := range r.Kvs {
-		found = strings.HasPrefix(string(v.Key), fmt.Sprintf("config/%d/receiver/%s", config.ProjectID, config.ID))
+		found = strings.HasPrefix(string(v.Key), ReceiverKey(config.ProjectID, config.ID))
 		if found {
 			assert.Equal(t, string(v.Value), encodedConfig)
 		}
 	}
 	assert.True(t, found, "inserted config not found")
 }
 
+func TestConfigStore_GetReceiver(t *testing.T) {
+	t.Parallel()
+
+	// Setup
+	ctx, d := newTestDeps(t)
+	store := NewConfigStore(d.logger, d.etcdClient, d.validator, d.tracer)
```

Comment:
Nie su tieto unit testy duplicitne s E2E testami?
Ako mozu zostat, ale dalsie duplicitne asi nemusime vytvarat.

Naopak, chyba mi tu unit test na `ReceiverTest` , `ProjectKey` a v `service` na `formatUrl`.

Reply:
Asi ano, ale tady jde spíš o to otestovat ten ConfigStore při developmentu "v izolaci". Nemusím dodělat zbytek (API, E2E testy) abych se ujistil, že tahle část funguje. Když už ty testy pak existují, tak je škoda je mazat.

U těch keys a formatUrl je to jen wrapper okolo formatování, aby se omylem někde nedal špatný format string a byly ty keys odlišné. Má to stejný účel jako vytvoření konstanty, takže mi přijde zbytečné to testovat

Reply:
Neviem co povie @JakubMatejka , ale ja by som povedal, ze chcem testovat aj tie dobne funkcie, a prave na to su dobre unit testy.

Ak ches pisat unit testy na to, co je uz pokryte E2E testami, ok, nevadi mi to.

Unit testy na tieto "drobne" funkcie su super v tom, ze ak nieco nejde, tak si mozes byt isty, ze chyba nie je v tej danej funkcii, a je to vidiet pekne v teste. A tak sa mozes focusovat na inec casti kodu, ako hladat chybu v `Sprintf`.

Reply:
Myslim, ze sa uplne bezne stane, ze urobis nejaky preklep aj v jednoduchej funkcii.
Okrem toho, ak by si mal chybu v nejakom `etcd` kluci, a dana funkcia sa pouziva aj na  create/update/delete/get, tak si chybu nenajdes, ale realne sa moze stat, ze ti tam bude chybat `projectId`, alebo zrobis nejaky BC break a testy ti prejdu.

Ak budes mat na to unit test, tak tam bude pekne vidiet, ze sa generovana hodnota zmenila, a ak si to nevsimnes ty, tak niekto iny na review.

Reply:
Mnó, tohle jsou už imho spíš jako funkční testy komunikace s databází než unit testy. (Resp. unit test na jednu funkci (`GetReceiver()`) by neměl používat jinou funkci z package (`CreateReceiver()`) na init té databáze.) Já je někdy psal ještě navíc k E2E testům, protože jo, Honza má pravdu že umí otestovat izolovaně tenhle package. Ale často se překrývají s E2E testy, mívají hodně podobný kód a je blbý když trvají dlouho. Tak je to spíš imho individuální otázka jestli mají dostatečnou přidanou hodnotu. A když teď děláme v podstatě CRUD API nad databází, tak je to asi spíš zbytečný.

Na druhou stranu unit testy jsou základ Keboolí Techové kultury a pokud je možný sebemenší funkci otestovat, tak by člověk měl cítit přímo nutkání test napsat. 😅 (Asi není třeba polemizovat k čemu jsou a proč je píšem, ne?)

Reply:
ok, mate pravdu, v tom balicku budu nakoniec aj ine veci, ako to co vidno cez API (len teraz, na zaciatku je to skoro `1:1` k tomu co sa da otestovat aj cez API).

Reply:
🤔 no já to myslel právě tak, že malý jednotkový testy na malý funkce (na `ProjectKey()` apod.) určitě ano, ale tady ty funkční testy možná nejsou potřeba, když nad těma funkcema je CRUD API který pokryjí E2E testy. Pokud ty API requesty budou dělat ještě další věci navíc než jen čistý CRUD nad jednou resoursou a/nebo bude v tomhle balíčku i další funkcionalita použitá jinde (což vlastně obojí platí když nad tím tak přemýšlím 😄), tak tyhle testy imho smysl dávají.

Reply:
ok, dohodneme sa na standupe v pondelok :) asi budeme musiet spravit aj nejaky testovaci helper co vypise vsetky kluce z `etcd` a to by mohli byt potom z toho unit testy, na kazdu funkciu samostatne.